src/plugins/discover/public/utils/format_value.ts

```diff
@@ -53,6 +53,13 @@ export function formatFieldValue(
       .convert(value, usedContentType, converterOptions);
   }
 
+  // format aggregate_metric_double field values
+  if (field.esTypes?.includes(ES_FIELD_TYPES.AGGREGATE_METRIC_DOUBLE)) {
```

Comment:
dear @ppisljar, have a question, you've been adding `ES_FIELD_TYPES.AGGREGATE_METRIC_DOUBLE` in https://github.com/elastic/kibana/pull/137677

It's assigned to `KBN_FIELD_TYPES.NUMBER`

So when we render it in Discover before this PR it's rendered without valid information, since field formatters try to render it like a number, but its data is an object. This PR catches this in Discover, renders it so it's useful. My question is now, is this approach fine? Since it's a special case only for Discover. Or should we e.g. teach field formatters to handle that case (a number is expected, but a object is given ... so format all field's values of that object as a number, return a string formatted like an object)

Reply:
I've discussed with @ppisljar and we think we should take one step further extending the `numeral` formatter to handle also object values:

https://github.com/elastic/kibana/blob/d6badd8715ecf37fb7899f85a671be48031e67bd/src/plugins/field_formats/common/converters/numeral.ts#L36

So all consumers of the field formatters would take advantage of that change.